app/lib/admin/metrics/measure/instance_accounts_measure.rb

```diff
@@ -45,11 +45,11 @@ def perform_data_query
         SELECT count(*) FROM new_accounts
       ) AS value
       FROM (
-        SELECT generate_series(date_trunc('day', $1::timestamp)::date, date_trunc('day', $2::timestamp)::date, interval '1 day') AS period
+        SELECT generate_series(date_trunc('day', :start_at::timestamp)::date, date_trunc('day', :end_at::timestamp)::date, interval '1 day') AS period
       ) AS axis
     SQL
 
-    rows = ActiveRecord::Base.connection.select_all(sql, nil, [[nil, @start_at], [nil, @end_at], [nil, params[:domain]]])
+    rows = Account.find_by_sql([sql, { start_at: @start_at, end_at: @end_at, domain: params[:domain] }])
```

Comment:
Seems kind of wasteful to instantiate `Account` for the results given that the results are not actually accounts.

Is there no better way?

Reply:
Oh wait -- I realize now you are referring to the instances of `Account` that get built, and not just using the `Account` class at all. I'll take a look to see if there's a way to comply w/ the Rails 7 changes w/out those instances.

Reply:
Refactored the whole thing to use an approach similar to https://github.com/mastodon/mastodon/pull/24249/files which means we can keep `select_all` and not load AR instances.

While in there I realized that many of the other classes which I hadn't changed had the same issue but it wasn't getting surfaced in the Rails 7 branch because their sql was not actually getting exercised by the spec suite. I added some specs to ensure that the data methods on all the measurement classes can at least run without erroring.